include/intel/cpu_support.h

```diff
@@ -0,0 +1,384 @@
+    alias_load<T>(make_zmm(zmm2) | mask, ptr[r_left_ + r9 * sizeof(T)]);
+    alias_load<T>(make_zmm(zmm1) | mask, ptr[r_right + r9 * sizeof(T)]);
+  }
+
+  template <class Operator,
+            Verify<Operator, op_ns::Mul, supported_types> = true>
+  void remainder_operations(const Xbyak::Opmask mask) {
+    typedef typename Operator::type IType;
+    remainder_fetch_LR<IType>(mask);
+    alias_MUL<IType>(zmm2, zmm2, zmm1);
+  }
+
+  template <class Operator,
+            Verify<Operator, op_ns::Add, supported_types> = true>
+  void remainder_operations(const Xbyak::Opmask mask) {
+    typedef typename Operator::type IType;
+    remainder_fetch_LR<IType>(mask);
+    alias_ADD<DType>(zmm2, zmm2, zmm1);
+  }
+
+  template <class Operator,
+            Verify<Operator, op_ns::Div, supported_types> = true>
+  void remainder_operations(const Xbyak::Opmask mask) {
+    typedef typename Operator::type IType;
+    remainder_fetch_LR<IType>(mask);
+    alias_DIV<DType>(zmm2, zmm2, zmm1);
+  }
+
+  template <class Operator,
+            Verify<Operator, op_ns::Sub, supported_types> = true>
+  void remainder_operations(const Xbyak::Opmask mask) {
+    typedef typename Operator::type IType;
+    remainder_fetch_LR<IType>(mask);
+    alias_SUB<DType>(zmm2, zmm2, zmm1);
+  }
+
+  elem_wise_add_update()
+      : r_out_(rdi),
+        r_left_(rsi),
+        r_right(rdx),
+        r_size_(rcx),
+        applicable_(false) {
+    static Xbyak::util::Cpu current_cpu;
+
+    /* Default case for all */
+    if (current_cpu.has(Xbyak::util::Cpu::tAVX512F)) {
+      /* prepare REMAINDER */
+      mov(r8, r_size_);
+      and_(r8,
+           UNIT_PER_REG - 1);  // r8_modulo = size/(sizeof(zmm)/sizeof(float))
+      xor_(r9, r9);            // reset r9
+      cmp(r_size_, UNIT_PER_REG);  // if ( size < 16 ) {  }
+      jl("remainder");
+
+      /*  decrease  divident */
+      sub(r_size_, r8);  // prepare alignment chunks
+      cmp(r_size_, 0);   // do we have any full chunks ?
+      jz("remainder");
+
+      L("for_i");
+      full_chunk_loop_operations<Op>();
+      add(r9, UNIT_PER_REG);  // r9+=sizeof(zmm)/sizeof(float)
+      cmp(r_size_, r9);       // more full chunks ?
+      jnz("for_i");
+
+      L("remainder");
+      cmp(r8, 0);  //  do we have a remainder ?
+      jz("done");
+      /* prepare a bitmask for k1 */
+      mov(rax, 1);
+      mov(r_size_, r8);
+      sal(rax, cl);
+      dec(rax);        // k1= (1 << r8 )-1
+      kmovw(k1, eax);  // set bitmask
+      alias_load<DType>(make_zmm(zmm0) | k1,
+                        ptr[r_out_ + r9 * UNIT_SIZE_BYTES]);
+      remainder_operations<Op>(k1);
+      alias_ADD<DType>(zmm3, zmm2, zmm0);
+      alias_save<DType>(ptr[r_out_ + r9 * UNIT_SIZE_BYTES],
+                        make_zmm(zmm3) | k1);
+      L("done");
+      applicable_ = true;
+      log_intel("AVX512F cpu kernel is ready");
+    }
+    ret();
+  }
+
+  bool is_applicable() const { return applicable_; }
```

Comment:
`is_applicable()` -> `applicable()`. For sheer member getter, use the same name.